compiler/rustc_serialize/src/opaque.rs

```diff
@@ -446,7 +448,11 @@ impl serialize::Encoder for FileEncoder {
 
     #[inline]
     fn emit_u16(&mut self, v: u16) -> FileEncodeResult {
-        file_encoder_write_leb128!(self, v, u16, write_u16_leb128)
+        let bytes = v.to_le_bytes();
+        unsafe {
+            self.write_one(*bytes.get_unchecked(0))?;
+            self.write_one(*bytes.get_unchecked(1))
+        }
```

Comment:
The `unsafe` code seems unnecessary. Why not just use `self.write_all(bytes)`?

Reply:
You're right, the code definitely deserves refactoring. I just wanted to see if there is any perf. benefit to it first, and I wanted to avoid any potential performance pitfalls for that.